modules/mvt/src/lib/types.ts

```diff
@@ -1,44 +1,44 @@
 export type MvtOptions = {
-  coordinates: string | any[];
+  coordinates: string | number[];
   tileIndex: {x: number; y: number; z: number};
   layerProperty: string | number;
-  layerName: any;
+  layerName: string;
 };
 
 export type MvtBinaryGeometry = {
   data: number[];
-  lines: number[];
+  lines: any[];
+  areas?: number[];
   type?: string;
-  id?: any;
+  id?: number;
 };
 
 export type MvtMapboxGeometry = {
   type?: string;
-  id?: any;
+  id?: number;
   length: number;
-  coordinates: any;
+  coordinates?: any[];
 };
 
 export type MvtBinaryCoordinates = {
   type: string;
   geometry: MvtBinaryGeometry;
-  properties: {};
-  id?: any;
+  properties: {[x: string]: string | number | boolean | null};
+  id?: number;
 };
 
 export type MvtMapboxCoordinates = {
   type: string;
   geometry: {
-    type?: string;
-    id?: any;
-    coordinates: any;
+    type: string;
+    coordinates: MvtMapboxGeometry;
   };
-  properties: {};
-  id?: any;
+  properties: {[x: string]: string | number | boolean | null};
+  id?: number;
 };
 
 export type MvtBinaryOptions = {
-  numericPropKeys: string | string[];
+  numericPropKeys: string[];
```

Comment:
Make sure you look at the binary geometry types in `@loaders.gl/schema`. 

Ultimately the idea is that these loaders return data in those standardized types.